packages/devtools_app/test/framework/observer/memory_observer_test.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2025 The Flutter Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file or at https://developers.google.com/open-source/licenses/bsd.
+
+import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_app/src/framework/observer/memory_observer.dart';
+import 'package:devtools_app/src/shared/feature_flags.dart';
+import 'package:devtools_app/src/shared/analytics/constants.dart' as gac;
+import 'package:devtools_app/src/shared/primitives/byte_utils.dart';
+import 'package:devtools_app_shared/utils.dart';
+import 'package:devtools_shared/devtools_test_utils.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  group('$MemoryObserver', () {
+    late MemoryObserver observer;
+
+    int memoryUsageBytes = 0;
+
+    Future<int?> testMeasureMemoryUsage() async => memoryUsageBytes;
+
+    setUp(() {
+      FeatureFlags.memoryObserver = true;
+      observer = MemoryObserver(
+        debugMeasureUsageInBytes: testMeasureMemoryUsage,
+        debugPollingDuration: const Duration(seconds: 2),
+      );
+      setGlobal(BannerMessagesController, BannerMessagesController());
+    });
+
+    tearDown(() {
+      observer.dispose();
+    });
+
+    test(
+      'does not add banner message when memory limit is not reached',
+      () async {
+        memoryUsageBytes =
+            convertBytes(1, from: ByteUnit.gb, to: ByteUnit.byte).round();
+        observer.init();
+        await delay(duration: const Duration(seconds: 3));
```

Comment:
nit: reduce this delay if possible (or better remove it, but that might be hard here).

These kinds of delays tend to add up quickly and result in long presubmit delays over time